tests/test_detectAndMeasure.py

```diff
@@ -312,28 +323,57 @@ def test_sky_sources(self):
-        self.assertLess(min(distance), matchDistance)
-        src = refSources[np.argmin(distance)]
-        if refIds is not None:
-            # Check that the same source was not previously associated
-            self.assertNotIn(src.getId(), refIds)
-            refIds.append(src.getId())
-        if atol is None:
-            atol = rtol*src.getPsfInstFlux() if usePsfFlux else rtol*src.getApInstFlux()
-        if usePsfFlux:
-            self.assertFloatsAlmostEqual(src.getPsfInstFlux()*scale, diaSource.getPsfInstFlux(),
-                                         rtol=rtol, atol=atol)
-        else:
-            self.assertFloatsAlmostEqual(src.getApInstFlux()*scale, diaSource.getApInstFlux(),
-                                         rtol=rtol, atol=atol)
+        # Set up the simulated images
+        noiseLevel = 1.
+        staticSeed = 1
+        transientSeed = 6
+        fluxLevel = 500
+        radius = 2
+        kwargs = {"seed": staticSeed, "psfSize": 2.4, "fluxLevel": fluxLevel}
+        science, sources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=6, **kwargs)
+        matchedTemplate, _ = makeTestImage(noiseLevel=noiseLevel/4, noiseSeed=7, **kwargs)
+
+        _checkMask = subtractImages.AlardLuptonSubtractTask._checkMask
+        # Configure the detection Task
+        detectionTask = self._setup_detection()
+        badMaskPlanes = detectionTask.config.detection.badMaskPlanes
+        nBad = len(badMaskPlanes)
+        self.assertGreater(nBad, 0)
+        kwargs["seed"] = transientSeed
+        kwargs["nSrc"] = nBad
+        kwargs["fluxLevel"] = 1000
+
+        # Run detection and check the results
+        def _detection_wrapper(setFlags=True):
+            transients, transientSources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=8, **kwargs)
+            difference = science.clone()
+            difference.maskedImage -= matchedTemplate.maskedImage
+            difference.maskedImage += transients.maskedImage
+            if setFlags:
+                for src, badMask in zip(transientSources, badMaskPlanes):
+                    srcX = int(src.getX())
+                    srcY = int(src.getY())
+                    srcBbox = lsst.geom.Box2I(lsst.geom.Point2I(srcX - radius, srcY - radius),
+                                              lsst.geom.Extent2I(2*radius + 1, 2*radius + 1))
+                    difference[srcBbox].mask.array |= lsst.afw.image.Mask.getPlaneBitMask(badMask)
+            output = detectionTask.run(science, matchedTemplate, difference)
+            refIds = []
+            goodSrcFlags = _checkMask(difference.mask, transientSources, badMaskPlanes)
+            if setFlags:
+                self.assertEqual(np.sum(~goodSrcFlags), nBad)
+            else:
+                self.assertEqual(np.sum(~goodSrcFlags), 0)
+            for diaSource, goodSrcFlag in zip(output.diaSources, goodSrcFlags):
+                if ~goodSrcFlag:
```

Comment:
What type is `goodSrcFlag` here, and why `~` instead of `not`?  I'm used to seeing that elementwise inversion of numpy arrays, but you also can't test for the truthiness of an entire array like this, so I wonder if you really mean `if not goodSrcFlag.any()` here.

Reply:
This test is a little simpler than that: it's iterating through all of the sources, so `goodSrcFlag` is a single boolean. I could certainly change that to `not goodSrcFalg` for readability.